src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] arrayOfElements;
+    private int size = 0;
+
+    public ArrayList() {
+        this.arrayOfElements = new Object[DEFAULT_CAPACITY];
+    }
+
+    private Object[] growIfArrayFull() {
+        Object[] grownArray = arrayOfElements;
+        if (size == arrayOfElements.length) {
+            grownArray = new Object[(int) (arrayOfElements.length * 1.5)];
+            System.arraycopy(arrayOfElements, 0, grownArray, 0, size);
+        }
+        return grownArray;
+    }
+
+    private void checkForValidIndexValue(int index) {
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid value of index.");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index + ". Size: " + size);
```
![image](https://user-images.githubusercontent.com/20326858/227191452-9b79b41b-eb1b-4b28-8a48-a1f301961686.png)